src/libraries/System.Private.CoreLib/src/System/DefaultBinder.cs

```diff
@@ -60,7 +60,7 @@ internal partial class DefaultBinder : Binder
                 if (names == null)
                 {
                     // Default mapping
-                    for (j = 0; j < args.Length; j++)
+                    for (j = 0; j < par.Length; j++)
                         paramOrder[i][j] = j;
```

Comment:
`paramOrder[i][j]` array keeps track of real(defined) position of parameters, in case `args` count is less than defined parameters its not tracked, in order to support any number of named parameters defined in any order we need to keep track of all parameters